shared/helpers/puppeteer.js

```diff
@@ -111,49 +125,94 @@ async function render(page, componentName, renderOptions, browserOptions) {
   // `evaluate`, only a `name` that maps to the error class (and not its `name`
   // property, which means we get a mangled value). As a workaround, we can
   // gather and `return` the values we need from inside the browser
-  const error = await page.evaluate(
-    async (selector, exportName, config) => {
-      const namespace = await import('govuk-frontend')
-
-      // Find all matching modules
-      const $modules = document.querySelectorAll(selector)
-
-      try {
-        // Loop and initialise all $modules or use default
-        // selector `null` return value when none found
-        ;($modules.length ? $modules : [null]).forEach(
-          ($module) => new namespace[exportName]($module, config)
-        )
-      } catch ({ name, message }) {
-        return { name, message }
-      }
-    },
-    selector,
-    exportName,
-    browserOptions?.config
-  )
-
-  // Throw Puppeteer errors back to Jest
-  if (error) {
-    throw new Error(
-      `Initialising \`new ${exportName}()\` with example '${exampleName}' threw:` +
-        `\n\t${error.name}: ${error.message}`,
-      { cause: error }
+  try {
+    const error = await page.evaluate(
+      async (selector, exportName, config) => {
+        const namespace = await import('govuk-frontend')
+
+        // Find all matching modules
+        const $modules = document.querySelectorAll(selector)
+
+        try {
+          // Loop and initialise all $modules or use default
+          // selector `null` return value when none found
+          ;($modules.length ? $modules : [null]).forEach(
+            ($module) => new namespace[exportName]($module, config)
+          )
+        } catch ({ name, message }) {
+          return { name, message }
+        }
+      },
+      selector,
+      exportName,
+      browserOptions?.config
     )
+
+    // Throw Puppeteer errors back to Jest
+    if (error) {
+      throw new Error(
+        `Initialising \`new ${exportName}()\` with example '${exampleName}' threw:` +
+          `\n\t${error.name}: ${error.message}`,
+        { cause: error }
+      )
+    }
+
+    await page.evaluateHandle('document.fonts.ready')
```

Comment:
I've got this comment in the commit message:

>Since fonts are downloaded asynchronously, we should await the `document.fonts.ready` promise since further network requests may not be available once the page has loads

This is because we turn the assets middleware on and off using:

```
page.on('request', middleware)
page.off('request', middleware)
```

Network requests can continue after the `render()` Promise resolves

>Similarly, some components are made visible after initialising JavaScript (or new elements created and inserted) so we should wait for additional font downloads to complete there too
>
>E.g. GDS Transport Light may have loaded already, but GDS Transport Bold may be requested on demand as required

You'll have noticed this running with `HEADLESS=false` on the Accordion examples. Since Accordion sections are not visible during page load only the **GDS Transport** Bold font is loaded for the headings

Accordion sections opened by tests appeared in Arial (with an HTTP 404 for **GDS Transport** Light)

Reply:
Cool, I think that makes things clearer for me, thanks for the extra detail. Should have mentioned that I had seen the comit but was really puzzled as to why it affected us more now than in our previous tests. Sounds like it's to ensure that `page.off` happens only after fonts are requested so that their request go through our request handler, which is very fine 👍🏻

Reply:
Brill. Similarly it will have happened when Express.js was torn down and test pages were still loading